src/navigation/navigation-events.ts

```diff
@@ -80,13 +91,18 @@ export function callCapturedEventListeners(eventArguments) {
   }
 }
 
-let urlRerouteOnly;
+let urlRerouteOnly: boolean;
 
-function urlReroute() {
-  reroute([], arguments);
+function urlReroute(evt: HashChangeEvent | PopStateEvent) {
+  reroute([], [evt]);
```

Comment:
This is a runtime change that I hope doesn't cause any changes to behavior for single-spa users. I made the change so that type checks were stricter.

The urlReroute function is added as an event listener to the window `hashchange` and `popstate` events. According to [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event), there is only one argument passed to hashchange and popstate events, so I don't think this change is safe.